File: Python/section1.2/friday.py
"""
ID: jameszh1
LANG: PYTHON2
TASK: friday
"""
def is_leap_year(year):
    if ((year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)):
        return True
    else:
        return False

def computeDaysOfMonth(month, year):
    if month in (4, 6, 9, 11):
        return 30
    if month == 2:
        if (is_leap_year(year)):
            return 29
        else:
            return 28
    return 31

INITIAL_YEAR = 1900
MONTH_NUM = 12
fin = open ('friday.in', 'r')
fout = open ('friday.out', 'w')
years = int(fin.readline().strip())
day_list = [0, 0, 0, 0, 0, 0, 0]
total_days = 0
for i in range(years):
    current_year = INITIAL_YEAR + i
    for j in range(1, MONTH_NUM + 1, 1):
        day_list[(total_days + 13) % 7] = day_list[(total_days + 13) % 7] + 1
        days_of_month = computeDaysOfMonth(j, current_year)
        total_days = total_days + days_of_month

fout.write (str(day_list[6]) + " " + str(day_list[0]) + " " + str(day_list[1]) + " " + str(day_list[2]) + " " + str(day_list[3]) + " " + str(day_list[4]) + " " + str(day_list[5]) + '\n')
fout.close()

File: src/section1_2/friday.java
/*
ID: jameszh1
LANG: JAVA
TASK: friday
*/

import java.io.*;
public class friday {

    public static void main(String[] args) throws Exception {
        int count[] = new int[7];
        BufferedReader br = new BufferedReader(new FileReader("friday.in"));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new FileWriter("friday.out")));
        int years = Integer.parseInt(br.readLine());
        int allDays = 0;
        for (int i = 1900; i < 1900 + years; i++) {
            for (int j = 1; j <= 12; j++) {
                count[(allDays + 13) % 7]++;
                allDays += countDaysInMonth(j, i);
            }
        }
        pw.println(count[6] + " " + count[0] + " " + count[1] + " " + count[2] + " " + count[3] + " " + count[4] + " " + count[5]);
        pw.close();
        br.close();
    }
    public static boolean isLeapYear(int year) {
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }else {
            return false;
        }
    }
    public static int countDaysInMonth(int month, int year) {
        switch (month) {
            case 4:
            case 6:
            case 9:
            case 11:
                return 30;
            case 2:
                if (isLeapYear(year)) {
                    return 29;
                }else {
                    return 28;
                }
            default:
                return 31;
        }
    }
}

